fe/fe-core/src/main/java/org/apache/doris/analysis/SortInfo.java

```diff
@@ -271,6 +271,11 @@ public ExprSubstitutionMap createSortTupleInfo(
      */
     public ExprSubstitutionMap createMaterializedOrderExprs(
             TupleDescriptor sortTupleDesc, Analyzer analyzer) {
+        // the sort node exprs may come from the child outer join node
+        // we need change the slots to nullable from all outer join nullable side temporarily
+        // then the sort node expr would have correct nullable info
+        // after create the output tuple we need revert the change by call analyzer.changeSlotsToNotNullable(slots)
+        List<SlotDescriptor> slots = analyzer.changeSlotToNullableOfOuterJoinedTuples();
```

Comment:
nit: maybe we could wrap these two helper function into a closable resource and use try statement to do auto change back to avoid unpaired use